src/views/PatronsPreRegistrationListContainer/hooks/useCreateNewUser.js

```diff
@@ -0,0 +1,47 @@
+import { useMutation } from 'react-query';
+import { useHistory } from 'react-router-dom';
+import { v4 as uuidv4 } from 'uuid';
+import { FormattedMessage } from 'react-intl';
+
+import { useOkapiKy, useCallout } from '@folio/stripes/core';
+
+import { PATRON_PREREGISTRATIONS_API, PERMISSIONS_API } from '../../../constants';
+
+// this hook is used to create a new user from pre-registration record
+const useCreateNewUser = () => {
```

Comment:
In my opinion, this hook takes on too much responsibility, which violates the "S" principle in SOLID. In addition to creating a user, it also manages navigation and displays messages in the callout. I would rather extract the direct creation of a new user for possible reuse. If you think this is excessive, feel free to ignore this comment.

Reply:
updated